src/backend/utils/misc/guc-file.l

```diff
@@ -472,6 +471,17 @@ ProcessConfigFileInternal(GucContext context, bool applySettings, int elevel)
 									item->name, item->value)));
 			}
 			item->applied = true;
+
+			if (changed_gucs_out)
+			{
+				struct config_generic *gconf = find_option(item->name, false, elevel);
+				if (gconf != NULL)
+				{
+					MemoryContext old_cxt = MemoryContextSwitchTo(caller_cxt);
```

Comment:
Why switching contexts is deemed necessary here?

Reply:
Because of [this](https://github.com/arenadata/gpdb/blob/446ca02da5bbeedc7c95e0173a11b5a1e7275ab9/src/backend/utils/misc/guc-file.l#L158). In GPDB 7 the whole `ProcessConfigFileInternal` function is called inside a separate memory context to avoid memory leaks. However if we want to return a list we have to allocate it in the caller's context so that it doesn't get destroyed at function return (or is it possible to do so without a full switch?)